### PR TITLE
[1.11 Backport]: fix(labelsReducer): fix labels reducer to accept empty strings

### PR DESCRIPTION
**Pull request by Poltergeist**
This commit is fixing the labels reducer to support labels with an empty string value.

Closes DCOS-19693

### CODE REVIEW

**Comment by nLight** [comment]
⚔️  flicts!

**Comment by juliangieseke** [comment]
@Poltergeist can you check the failed test? could be a legit one 🤔

File: tests/pages/services/ServiceReviewScreen-cy.js [hunk 1]
```diff
@@ -1197,6 +1197,7 @@ describe("Services", function() {
                 mode: "RW",
                 containerPath: "test"
               }
+
```
Comment (juliangieseke) [inline tests/pages/services/ServiceReviewScreen-cy.js]: 🤔

File: plugins/services/src/js/components/modals/CreateServiceModalForm.js [hunk 2]
```diff
@@ -309,7 +319,18 @@ class CreateServiceModalForm extends Component {
       baseConfigCopy
     );
 
-    return CreateServiceModalFormUtil.stripEmptyProperties(newConfig);
+    const { labels, ...config } = newConfig;
+
+    if (Object.keys(labels).length === 0) {
+      return {
```
Comment (juliangieseke) [inline plugins/services/src/js/components/modals/CreateServiceModalForm.js]: shouldnt be `return CreateServiceModalFormUtil.stripEmptyProperties(config)` be enough here?

File: plugins/services/src/js/components/modals/CreateServiceModalForm.js [hunk 3]
```diff
@@ -309,7 +319,18 @@ class CreateServiceModalForm extends Component {
...
+        ...CreateServiceModalFormUtil.stripEmptyProperties(config)
+      };
+    }
+
+    return {
+      labels,
```
Comment (juliangieseke) [inline plugins/services/src/js/components/modals/CreateServiceModalForm.js]: why are we destructing it above, only to merge it together again? Is this a way to fix a broken `CreateServiceModalFormUtil.stripEmptyProperties`?

Comment (juliangieseke) [reply plugins/services/src/js/components/modals/CreateServiceModalForm.js]: `return CreateServiceModalFormUtil.stripEmptyProperties(newConfig)`?!

**Comment by Poltergeist** [comment]
Alfred you can't test this. 👯 👯‍♂️ 🎉

**Comment by ahoskins** [comment]
Alfred test this please

**Comment by nLight** [comment]
Alfred test this

**Comment by nLight** [comment]
Alfred test this

**Comment by nLight** [comment]
Alfred test this

**Comment by nLight** [comment]
Alfred test this

**Comment by nLight** [comment]
Alfred test this

**Pull request merged**